lib/transformer.dart

```diff
@@ -0,0 +1,399 @@
+library w_module.transformer;
+
+import 'package:analyzer/analyzer.dart';
+import 'package:barback/barback.dart';
+import 'package:source_span/source_span.dart';
+
+import 'package:w_module/annotations.dart' as annotations;
+
+import 'package:w_module/src/transformer/utils.dart';
+
+/// Aggregate transformer that generates the implementation logic for a deferred
+/// module. The generated implementation takes care of the following:
+///
+/// - Loading the deferred library that contains the real module class when the
+///   deferred module's "onLoad" event occurs.
+/// - Deferring the availability of API, Components, and Events on the deferred
+///   module until the underlying module has actually been loaded. In other
+///   words, attempting to access the API, Components, or Events will throw if
```

Comment:
I think we talked about it briefly before, but what are your thoughts on automatically loading the module on any attempted access of the API, Components, or Events?  instead of throwing, if the deferred module hasn't yet been loaded, await load before proceeding with the call (might have to guard against multiple async calls inadvertently triggering load...)